// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package pb;


// The request message containing the user's name.
message BookRequest {
  string name = 2;
  int64 size = 3;
}

message Books{
  int64 id = 1;
  string name = 2;  
  float price = 3;
  repeated string image = 5;
  string description = 4;
  string author = 6;
  string publisher = 7;
  int32 quantity = 8;
  float rating = 9;
  bool is_deleted = 10;
}

message BookResponse {
  repeated Books books = 1;
}

service BookRecommend {
  rpc GetBookRecommend (BookRequest) returns (BookResponse) {};
}